poetry/installation/pip_installer.py

```diff
@@ -193,7 +194,7 @@ def install_directory(self, package):
             # We also need it for non-PEP-517 packages
             builder = SdistBuilder(Poetry.create(pyproject.parent), NullEnv(), NullIO())
 
-            with open(setup, "w", encoding="utf-8") as f:
```

Comment:
`encoding` is not a valid param in python 2 `open`. Both Python 2 and Python 3 have an `open` function in the `io` module that is actually the same as the builtin Python 3 `open`.